src/main/java/frc/robot/subsystems/scoring/ScoringSubsystem.java

```diff
@@ -100,16 +128,41 @@ private void endgame() { // TODO: Later
         state = ScoringState.IDLE;
     }
 
-    private double findShootAngleRads() { // TODO: Interpolate
-        double distancetoGoal = Math.sqrt(Math.pow(Math.sqrt(Math.pow(1, 2) + Math.pow(1, 2)), 2) + Math.pow(1, 2));
-        return Math.atan2(distancetoGoal, 1);
+    private double findDistanceToGoal() {
+        Translation2d speakerPose =
+                false // TODO: CHANGE THIS URGENT
+                        // DriverStation.getAlliance().get() == DriverStation.Alliance.Red
+                        ? new Translation2d(
+                                Units.inchesToMeters(652.73), Units.inchesToMeters(218.42))
+                        : new Translation2d(
+                                Units.inchesToMeters(-1.5),
+                                Units.inchesToMeters(218.42)); // TODO: Might have to change these
+        Pose2d robotPose = poseSupplier.get();
+        double distancetoGoal =
+                Math.sqrt(
+                        Math.pow(Math.abs(robotPose.getX() - speakerPose.getX()), 2)
+                                + Math.pow(
+                                        Math.abs(robotPose.getY() - speakerPose.getY()),
+                                        2)); // TODO: Change to be more accurate
```

Comment:
what is more accurate than euclidean distance?

Reply:
Sorry, that was leftover from when the speaker's pose was 0,0.